pkg/storage/pebble_key_schema_test.go

```diff
@@ -312,3 +315,38 @@ func parseTestKey(s string) ([]byte, error) {
 		}), nil
 	}
 }
+
+func TestKeySchema_RandomKeys(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	rng, _ := randutil.NewTestRand()
+	maxUserKeyLen := randutil.RandIntInRange(rng, 2, 10)
+	keys := make([][]byte, randutil.RandIntInRange(rng, 1, 1000))
+	for i := range keys {
+		keys[i] = randomSerializedEngineKey(rng, maxUserKeyLen)
+	}
+	slices.SortFunc(keys, EngineKeyCompare)
+
+	var enc colblk.DataBlockEncoder
+	enc.Init(keySchema)
+	for i := range keys {
+		ikey := pebble.InternalKey{
+			UserKey: keys[i],
+			Trailer: pebble.MakeInternalKeyTrailer(0, pebble.InternalKeyKindSet),
+		}
+		enc.Add(ikey, keys[i], block.InPlaceValuePrefix(false), enc.KeyWriter.ComparePrev(keys[i]), false /* isObsolete */)
+	}
+	blk, _ := enc.Finish(len(keys), enc.Size())
+	blk = crbytes.CopyAligned(blk)
+
+	var dec colblk.DataBlockDecoder
+	dec.Init(keySchema, blk)
+	var it colblk.DataBlockIter
+	it.InitOnce(keySchema, EngineKeyCompare, EngineKeySplit, nil)
+	require.NoError(t, it.Init(&dec, block.NoTransforms))
+	for k, kv := 0, it.First(); kv != nil; k, kv = k+1, it.Next() {
+		require.True(t, EngineKeyEqual(keys[k], kv.K.UserKey))
```

Comment:
Should we compare for exact equality? We assume that the keys will be identical (or at least never longer) when we preallocate the max-key-length buffer

Reply:
Added an assertion on the length. This test will generate MVCC keys with the synthetic bit so we can't assert exact equality